### PR TITLE
[artifactory][artifactory-ha] Add support for global values

### PR DESCRIPTION
**Pull request by chukka**
#### PR Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Chart Version bumped
- [x] CHANGELOG.md updated
- [x] Variables and other changes are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[artifactory]`)

**What this PR does / why we need it**:
[artifactory] Add support for global values

### CODE REVIEW

File: stable/artifactory/templates/_helpers.tpl [hunk 1]
```diff
@@ -83,3 +83,63 @@ Scheme (http/https) based on Access TLS enabled/disabled
...
+{{- define "artifactory.masterKey" -}}
+{{- default .Values.artifactory.masterKey .Values.global.masterKey -}}
+{{- end -}}
+
+
+{{/*
+Resolve joinKeySecretName value
+*/}}
+{{- define "artifactory.joinKeySecretName" -}}
+{{- default (include "artifactory.fullname" .) .Values.artifactory.joinKeySecretName .Values.global.joinKeySecretName -}}
```
Comment (danielezer) [inline stable/artifactory/templates/_helpers.tpl:106]: @chukka Can you explain the `include`? I don't understand its purpose here

File: stable/artifactory/templates/artifactory-statefulset.yaml [hunk 2]
```diff
@@ -214,8 +214,8 @@ spec:
           done;
       {{- end }}
     {{- end }}
-    {{- if .Values.artifactory.customInitContainers }}
-{{ tpl .Values.artifactory.customInitContainers . | indent 6 }}
+    {{- if or .Values.artifactory.customInitContainers }}
```
Comment (danielezer) [inline stable/artifactory/templates/artifactory-statefulset.yaml:217]: You are missing the global value here

**Review by danielezer — CHANGES_REQUESTED** [review]

File: stable/artifactory/templates/_helpers.tpl (hunk 1 above)
Comment (chukka) [reply stable/artifactory/templates/_helpers.tpl:106]: @danielezer  - https://github.com/jfrog/charts/blob/master/stable/artifactory/templates/artifactory-secrets.yaml#L4 - secrets from values.yaml

File: stable/artifactory/templates/artifactory-statefulset.yaml (hunk 2 above)
Comment (chukka) [reply stable/artifactory/templates/artifactory-statefulset.yaml]: Done

File: stable/artifactory/templates/_helpers.tpl (hunk 1 above)
Comment (danielezer) [reply stable/artifactory/templates/_helpers.tpl:106]: Can `default` accept three arguments? This looks weird to me

Comment (chukka) [reply stable/artifactory/templates/_helpers.tpl:106]: yes, it does, we are already using them - https://github.com/jfrog/charts/blob/master/stable/artifactory/templates/_helpers.tpl#L13

Comment (danielezer) [reply stable/artifactory/templates/_helpers.tpl:106]: Ok, cool. Good to know

**Review by rimusz — APPROVED** [review]
LGTM

**Review by rahulsadanandan — APPROVED** [review]

**Pull request merged**